java/languages.antlr/src/org/netbeans/modules/languages/antlr/AntlrLanguage.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.languages.antlr;
+
+import org.netbeans.api.lexer.Language;
+import org.netbeans.core.spi.multiview.MultiViewElement;
+import org.netbeans.core.spi.multiview.text.MultiViewEditorElement;
+import org.netbeans.modules.csl.api.DeclarationFinder;
+import org.netbeans.modules.csl.api.OccurrencesFinder;
+import org.netbeans.modules.csl.api.StructureScanner;
+import org.netbeans.modules.csl.spi.DefaultLanguageConfig;
+import org.netbeans.modules.csl.spi.LanguageRegistration;
+import org.netbeans.modules.parsing.spi.Parser;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.filesystems.MIMEResolver;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+import org.openide.windows.TopComponent;
+
+/**
+ *
+ * @author lkishalmi
+ */
+@NbBundle.Messages(
+        "ANTLRResolver=ANTLR Grammar"
+)
+@MIMEResolver.ExtensionRegistration(displayName = "#ANTLRResolver",
+    extension = {"g4", "g"},
+    mimeType = AntlrTokenId.MIME_TYPE,
+    position = 285
+)
```

Comment:
Please split the language definitions. Mapping antlr3 and antlr4 to the same mimetype, will make it impossible to hook on only one type of grammar. Given, that there was a reason, that antlr4 was created (I assume it is more expressive than antlr3), I can imagine such requirements. Setting the unificiation into stone will come to haunt us.

Reply:
Well, I've spent a lot of time on that during the weekend. I did not want to duplicate the editor preferences, so I've tried with ```text/x-antlr3+x-antlr``` and ```text/x-antlr4+x-antlr``` mime types but the CSL does not support complex mimetypes.

Will try to separate the two entirely.